core/src/services/fs/writer.rs

```diff
@@ -68,7 +68,7 @@ impl oio::Write for FsWriter<tokio::fs::File> {
     async fn sink(&mut self, _size: u64, mut s: oio::Streamer) -> Result<()> {
         while let Some(bytes) = s.next().await {
             let b = bytes?;
-            self.f.write_all(&b).await.map_err(parse_io_error)?;
+            self.f.write(b).await.map_err(parse_io_error)?;
         }
```

Comment:
We should count the size and update the position.

Reply:
My bad, i thought self.f was calling the function `write` in `oio::Writer`